rgf/test/test.py

```diff
@@ -73,6 +71,17 @@ def test_regressor(self):
         print("mse: " + str(mse))
         assert_less(mse, 6.0)
 
+    def test_sample_weight(self):
+        X_train, X_test, y_train, y_test = train_test_split(self.iris.data,
+                                                            self.iris.target,
+                                                            test_size=0.2,
+                                                            random_state=42)
+
+        clf = RGFClassifier()
+        y_pred = clf.fit(X_train, y_train).predict_proba(X_test)
+        y_pred_weighted = clf.fit(X_train, y_train, np.ones(y_train.shape[0])).predict_proba(X_test)
+        self.assertTrue(np.isclose(y_pred, y_pred_weighted).all())
```

Comment:
Unfortunately, it is impossible to perform your test because sample weights must be positive.
![image](https://cloud.githubusercontent.com/assets/25141164/26626453/f32ded4a-45ff-11e7-8479-7cef452e9788.png)